omise/controllers/front/payment.php

```diff
@@ -0,0 +1,28 @@
+<?php
+if (! defined('_PS_VERSION_')) {
+    exit();
+}
+
+if (defined('_PS_MODULE_DIR_')) {
+    require_once _PS_MODULE_DIR_ . '/omise/classes/charge.php';
+}
+
```

Comment:
ok, last question about this design (and it'd not relate with what we've talked above).

Because this design moved code into `Charge` controller, so, it hard to reuse the class (i.e. if we implement "Refund" service).

I'd suggest to 
- make `OmisePaymentModuleFrontController` as an abstract method that extend `ModuleFrontController`. 
- Then, `Charge` will extends `OmisePaymentModuleFrontController`.
- Then, we implement `postProcess` method in `Charge.php` (and update endpoint of submit payment to this class instead)

- Then, provide some basic functions that we could reuse in every controller classes to `OmisePaymentModuleFrontController`
i.e. 
```php
abstract class OmisePaymentModuleFrontController extends ModuleFrontController
{
    protected $setting;

    public function __construct()
    {
        $this->setting = new Setting();
    }

    public function redirect()
    {
        // Redirect to the result/ 3-D Secure/ Internet Banking page.
    }

    public function handleFailure()
    {
        // - Update order status to "failed",
        // - Redirect to fail page
        // - Etc.
    }
...
```

- Now you can use `$this->context` class inside `Charge`, and every controller class in the future can also extend `OmisePaymentModuleFrontController` to use `$this->context`.

- Then, in `Charge` Controller, just provide only method that can't reuse. (i.e. `$this->getCardToken()`)

- So, in `Charge` controller could be

```php
class Charge extends OmisePaymentModuleFrontController
{
    public function postProcess()
    {
        $charge_request = array(
            'amount'      => $this->getAmount(),
            'card'        => $this->getCardToken(),
            'capture'     => $this->getCapture(),
            'currency'    => $this->context->currency->iso_code ,
            'description' => $this->getChargeDescription(),
        );

        try {
            OmiseCharge::create($charge_request, '', $this->setting->getSecretKey());

            $this->setRedirectAfter('success-page');
        } catch(Exception $e) {
            $this->handleFailure();
        }
    }

...
```

☝️ I know this idea will not solve "decoupling" class issue as you want much.

But since PrestaShop has been provided `postProcess` for us, each class (i.e. `Charge.php`, `Refund.php`) just focus on implement their own `postProcess()` method, and lets handler methods live in `OmisePaymentModuleFrontController`

It'd make `OmisePaymentModuleFrontController` can be reusable and reduce duplicated code in case we have another controller for another service (i.e. refund).